### PR TITLE
miner

### PR DESCRIPTION
**Pull request by turuslan**
- miner main
- move `LocalStorage` from `Repository`
- fix `operator<`
- `MapPrefix`
- remove `Address` network field

### CODE REVIEW

File: test/core/sector_storage/stores/local_store_test.cpp [hunk 1]
```diff
@@ -578,8 +577,7 @@ TEST_F(LocalStoreTest, removeNotExistSector) {
       .WillOnce(
           testing::Return(fc::outcome::success(std::vector<StorageInfo>())));
 
-  EXPECT_OUTCOME_ERROR(StoreErrors::kNotFoundSector,
-                       local_store_->remove(sector, type));
+  EXPECT_OUTCOME_TRUE_1(local_store_->remove(sector, type));
```
Comment (artyom-yurin) [inline test/core/sector_storage/stores/local_store_test.cpp:580]: Test description should be changed

File: core/miner/impl/miner_impl.hpp [hunk 2]
```diff
@@ -53,7 +53,6 @@ namespace fc::miner {
 
     Address getAddress() const override;
 
-   private:
```
Comment (Alexey-N-Chernyshov) [inline core/miner/impl/miner_impl.hpp:56]: Making all members public violates the principle of information hiding.

File: core/storage/leveldb/leveldb.cpp [hunk 3]
```diff
@@ -20,12 +20,19 @@ namespace fc::storage {
     if (status.ok()) {
       auto l = std::make_shared<LevelDB>();
       l->db_ = std::unique_ptr<leveldb::DB>(db);
-      return std::move(l); // clang 6.0.1 issue
+      return std::move(l);  // clang 6.0.1 issue
```
Comment (Alexey-N-Chernyshov) [inline core/storage/leveldb/leveldb.cpp:23]: What kind of issue?

File: core/storage/ipfs/api_ipfs_datastore/api_ipfs_datastore.cpp [hunk 4]
```diff
@@ -13,9 +13,6 @@ namespace fc::storage::ipfs {
   outcome::result<bool> ApiIpfsDatastore::contains(const CID &key) const {
     auto res = api_->ChainReadObj(key);
     if (res.has_error()) {
-      if (res.error() == IpfsDatastoreError::kNotFound) {
```
Comment (Alexey-N-Chernyshov) [inline core/storage/ipfs/api_ipfs_datastore/api_ipfs_datastore.cpp:16]: If 'kNotFound' is treated as error, why return value is 'bool'? It is always 'true' or 'error'.

File: core/storage/leveldb/leveldb.cpp (hunk 3 above)
Comment (turuslan) [reply core/storage/leveldb/leveldb.cpp:23]: `clang-format` created that change, this isn't my code.

File: core/storage/ipfs/api_ipfs_datastore/api_ipfs_datastore.cpp (hunk 4 above)
Comment (turuslan) [reply core/storage/ipfs/api_ipfs_datastore/api_ipfs_datastore.cpp:16]: Rpc can't return `kNotFound`, so treating error consistently

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/miner/main/main.cpp [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    OUTCOME_TRY(node_peer, napi->NetAddrsListen());
+    host->connect(node_peer);
+
+    auto storage{std::make_shared<sector_storage::stores::LocalStorageImpl>(
+        config.join("storage.json"))};
+    OUTCOME_TRY(storage->setStorage([&](auto &config2) {
+      if (config2.storage_paths.empty()) {
+        boost::filesystem::path path{config.join("sectors")};
+        boost::filesystem::create_directories(path);
+        // TODO(artyom-yurin): check meta
```
Comment (artyom-yurin) [inline core/miner/main/main.cpp:232]: ?

**Review by artyom-yurin — APPROVED** [review]

**Pull request merged**